frontend/src/component/project/Project/PaginatedProjectFeatureToggles/hooks/useDefaultColumnVisibility.test.tsx

```diff
@@ -0,0 +1,74 @@
+import theme from 'themes/theme';
+import { screen } from '@testing-library/react';
+import { useDefaultColumnVisibility } from './useDefaultColumnVisibility';
+import { render } from 'utils/testRenderer';
+import { ThemeProvider } from 'themes/ThemeProvider';
+import mediaQuery from 'css-mediaquery';
+
+const createMatchMedia = (width: number) => {
+    return (query: string) => {
+        return {
+            matches: mediaQuery.match(query, { width }),
+            media: '',
+            addListener: () => {},
+            removeListener: () => {},
+            onchange: () => {},
+            addEventListener: () => {},
+            removeEventListener: () => {},
+            dispatchEvent: () => true,
+        };
+    };
+};
+
+const resizeScreen = (width: number) => {
+    window.matchMedia = createMatchMedia(width);
+};
+
+const columnIds = [
+    'select',
+    'favorite',
+    'name',
+    'createdAt',
+    'lastSeenAt',
+    'environment:development',
+    'environment:production',
+    'environment:dev2',
+    'environment:prod2',
+    'environment:staging',
+    'environment:test',
+    'actions',
+];
+
+const TestComponent: React.FC = () => {
+    const columns = useDefaultColumnVisibility(columnIds);
+
+    return (
+        <ThemeProvider>
+            <ul data-testid='wrapper'>
+                {Object.keys(columns).map((column) => (
+                    <li key={column}>{column}</li>
+                ))}
+            </ul>
+        </ThemeProvider>
+    );
+};
+
+test.each(Object.keys(theme.breakpoints.values))(
+    'it renders all envs on %s screens',
+    (screenSize) => {
+        resizeScreen(
+            theme.breakpoints.values[
+                screenSize as keyof typeof theme.breakpoints.values
+            ] + 1,
```

Comment:
I suspect we have an off-by-one issue here. If you cut a stick in 2 places, how many sticks you will end up with? It's the same with breakpoints.

Reply:
That's a fair point. But I had in fact thought of this. The thing is, assuming I'm looking at the right file (`frontend/src/themes/theme.ts`), then breakpoints is defined as such:

```ts
    breakpoints: {
        values: {
            xs: 0,
            sm: 600,
            md: 960,
            lg: 1280,
            xl: 1536,
        },
    },
```

So in this case, the smallest size we set for the screen is 1 pixel. The way I read this, `xs` is from 0 px and up, `sm` from 600 px and up etc. Thus, using one px above their lower threshold should put us safely within their zone.

Am I making sense?

Reply:
Or, the breakpoint values are defined by their lower bound.

Reply:
O, perfect, thanks for clarification

Reply:
In theory, we could probably skip the +1, but that just means we avoid any weird edge cases if we use strictly vs greater than or equal to